src/arrayMethodJoin.js

```diff
@@ -5,7 +5,27 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let joined = '';
+    let divider = separator;
+
+    if (separator === undefined) {
+      divider = ',';
+    }
```

Comment:
Instead of checking argument for undefined, you can set a default value to function parameter.
```suggestion
function(separator = ',')
```